src/rgw/rgw_admin.cc

```diff
@@ -10532,40 +10532,34 @@ int main(int argc, const char **argv)
 
   if (opt_cmd == OPT::PUBSUB_TOPIC_LIST) {
     RGWPubSub ps(driver, tenant);
-    rgw_pubsub_topics result;
-    int ret = ps.get_topics(dpp(), result, null_yield);
-    if (ret < 0 && ret != -ENOENT) {
-      cerr << "ERROR: could not get topics: " << cpp_strerror(-ret) << std::endl;
-      return -ret;
-    }
-    if (!rgw::sal::User::empty(user)) {
-      for (auto it = result.topics.cbegin(); it != result.topics.cend();) {
-        const auto& topic = it->second;
-        if (user->get_id() != topic.user) {
-          result.topics.erase(it++);
-        } else {
-          ++it;
-        }
+    std::string next_token;
+
+    formatter->open_object_section("result");
+    formatter->open_array_section("topics");
+    do {
+      rgw_pubsub_topics result;
+      int ret = ps.get_topics(dpp(), next_token, max_entries,
+                              result, next_token, null_yield);
+      if (ret < 0 && ret != -ENOENT) {
+        cerr << "ERROR: could not get topics: " << cpp_strerror(-ret) << std::endl;
+        return -ret;
       }
-    }
-    if (driver->get_zone()->get_zonegroup().supports_feature(
-            rgw::zone_features::notification_v2)) {
-      Formatter::ObjectSection top_section(*formatter, "result");
-      Formatter::ArraySection s(*formatter, "topics");
       for (const auto& [_, topic] : result.topics) {
+        if (!rgw::sal::User::empty(user) && user->get_id() != topic.user) {
+          continue;
+        }
         std::set<std::string> subscribed_buckets;
         ret = driver->get_bucket_topic_mapping(topic, subscribed_buckets,
                                                null_yield, dpp());
         if (ret < 0) {
           cerr << "failed to fetch bucket topic mapping info for topic: "
                << topic.name << ", ret=" << ret << std::endl;
-        } else {
-          show_topics_info_v2(topic, subscribed_buckets, formatter.get());
         }
+        show_topics_info_v2(topic, subscribed_buckets, formatter.get());
```

Comment:
this call and `get_bucket_topic_mapping()` should only be called for `v2`, and  if we v1 also should display an empty `subscribed_bucket` list, then we could rename the `show_topics_info_v2` to `show_topics_info`